modules/imgcodecs/src/grfmt_jpeg.cpp

```diff
@@ -242,9 +242,26 @@ bool  JpegDecoder::readHeader()
         {
             jpeg_read_header( &state->cinfo, TRUE );
 
-            state->cinfo.scale_num=1;
-            state->cinfo.scale_denom = m_scale_denom;
-            m_scale_denom=1; // trick! to know which decoder used scale_denom see imread_
+            int m_scale_denom = max( m_size.width , m_size.height );
+            if ( (m_size.width | m_size.height ) == 0 )
+            {
+                m_scale_denom = 1;
+            }
+            else if( m_scale_denom < 9 )
+            {
+                m_size = Size(); // trick! to know JpegDecoder used scale_denom see imread_
+            }
+            else if( m_scale_denom == m_size.width )
+            {
+                m_scale_denom = state->cinfo.image_width / m_size.width;
+            }
+            else if( m_scale_denom == m_size.height )
+            {
+                m_scale_denom = state->cinfo.image_height / m_size.height;
+            }
+
+            state->cinfo.scale_num = 1;
+            state->cinfo.scale_denom = ( m_scale_denom > 1 ) ? m_scale_denom : 1;
```

Comment:
which is better ?
state->cinfo.scale_denom = ( m_scale_denom > 1 ) ? m_scale_denom : 1;
state->cinfo.scale_denom = m_scale_denom ? m_scale_denom : 1;